### ISSUE TITLE
Is the doorman still part of Corda?

### ISSUE THREAD

**Issue by george-hawkins** [corda/corda#2703]
The current situation with the doorman is a bit confusing.

The doorman is clearly a key element of the [v1.0 network permissioning documentation](https://docs.corda.net/releases/release-V1.0/permissioning.html). However all mention of the doorman has been removed in the current corresponding [v2.0 documentation](https://docs.corda.net/permissioning.html) (and neither is there any mention of any kind of doorman-like service that's taken its place).

Two days ago, in the master branch of the main Corda project, reference to the doorman was reintroduced in the permissioning documentation in commit [`754b87d`](https://github.com/corda/corda/commit/754b87d547c2a7f737a18ae84b64d41947f0d420#diff-e0a688ae8fed154e0fbe8b5feff74419). Now though the doorman seems to be just a new name for the intermediate network CA, where users of Corda must manage the relevant keystore themselves, rather than a service as it seemed to be in the v1.0 documentation.

The [`HTTPNetworkRegistrationService`](https://github.com/corda/corda/blob/master/node/src/main/kotlin/net/corda/node/utilities/registration/HTTPNetworkRegistrationService.kt) class whose job it seems to have been to submit requests to the doorman service (and retrieve certificates back) still exists in the master branch.

So does the doorman as a service still exist, rather than just being another keystore representing a layer in the CA hierarchy (a layer that for some reason was for a while named the intermediate network CA)?

The sequence of changes here is quite confusing - what we currently see on the web as the [v2.0 network permissioning documentation](https://docs.corda.net/permissioning.html) makes no mention of the doorman, however it is not the version of the documentation that got tagged as release-V2.0 - the [release-V2.0 tagged version](https://github.com/corda/corda/blob/release-V2.0/docs/source/permissioning.rst) does mention the doorman!

Things don't look much clearer if I look at the log history of `permissioning.rst` - all references to the doorman were removed in commit [`20337aa`](https://github.com/corda/corda/commit/20337aa) on Nov 24, 2017 only to be reintroduced (as a CA keystore rather than a service) three months later in commit [`754b87d`](https://github.com/corda/corda/commit/754b87d547c2a7f737a18ae84b64d41947f0d420#diff-e0a688ae8fed154e0fbe8b5feff74419).

And as noted it's not the release-V2.0 tag in which we see first see reference to the doorman being removed but in the later [ENT-HC01 tag](https://github.com/corda/corda/blob/ENT-HC01/docs/source/permissioning.rst).

In short - some clarification here would be appreciated. Various pages in the current v2.0 documentation on the web refer to the doorman, e.g. the certificates section of the [identity page](https://docs.corda.net/key-concepts-identity.html) refers to the doorman and links to the permissioning page discussed above, but to a version of the page where reference to the doorman have been removed! Similarly, as noted above, code to access the doorman service, e.g. the `HTTPNetworkRegistrationService`, still seems to exist - however the doorman references introduced recently to the permissioning document no longer describe it as a service.

Perhaps @rnicoll, @patrickkuo or @shamsasari would like to comment on what's going on here or where things are headed. Thanks :smile:

**Comment by mikehearn** [comment]
Hi George, sorry for the confusion around this. The churn you've seen is because we hadn't finished all the network protocols in version 1/2, so this stuff has only been consolidating for version 3. The term "doorman" refers to an HTTP server that accepts certificate signing requests and signs node certs if the operator of the server wants to welcome that node into their compatibility zone (you can think of a zone as a permissioned network). In v3 we have documented this protocol but don't provide a built in doorman server. This is because such servers tend to consist 90% of zone-specific business logic, like what sort of names do you want to allow, integration with a ticketing system  (if any) for manual review and so on.

In R3 Corda, our upcoming commercial version, we do provide our in-house R3 doorman as a base for customisation or adoption if you want to use the same infrastructure or policies as us. Or you could implement your own version of this protocol.

But the easiest thing to do is wait for us (the Corda team) to formally launch our own compatibility zone. The goal is to have as many users as possible in this zone so all the nodes interoperate successfully, which includes the shared identity directory that the network map service provides. We'll be talking more about this at CordaCon and will be making more announcements about this in future.

**Comment by george-hawkins** [comment]
Cool - thanks Mike for that feed back. PS on a different topic I logged some issues ([#12](https://github.com/corda/corda-gradle-plugins/issues/12) and [#18](https://github.com/corda/corda-gradle-plugins/issues/18)) against the corda-gradle-plugins project but they seem to be just sitting there quietly (maybe bugs are just assigned on a periodic basis?). As I've fixed them in my local code base they're not immediate issues for me.

**Comment by mikehearn** [comment]
The maintainer of that module has been on holiday. He'll be back next week. We're also going to set up more sub-maintainers to ensure that bugs and PRs get looked at faster. I used to do it but I'm pretty busy with other things these days, so it's time to scale the open source project!
